tests/ci/run_m1_ec2_instance.sh

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+set -exo pipefail
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR ISC
+
+# Please run from project root folder!
+# You'll want to set the codebuild env variables set if running locally
+source tests/ci/common_ssm_setup.sh
+
+# cleanup code
+cleanup() {
+  set +e
+  # delete the various documents that we created
+  for name in ${ssm_document_names};do
+    aws ssm delete-document --name "${name}"
+  done
+}
+
+# we wanna run the cleanup code on exit
+trap cleanup EXIT
+
+# print some information for reference
+echo GitHub PR Number: "${CODEBUILD_WEBHOOK_TRIGGER}"
+echo GitHub Branch Name: "${CODEBUILD_WEBHOOK_HEAD_REF}"
+AWS_ACCOUNT_ID=$(aws sts get-caller-identity --query Account --output text)
+echo AWS Account ID: "${AWS_ACCOUNT_ID}"
+echo GitHub Repo Link: "${CODEBUILD_SOURCE_REPO_URL}"
+export cloudwatch_group_name="aws-lc-ci-macos-arm-cw-logs"
+
+# get information for ec2 instances
+ec2_instance="$(aws ec2 describe-instances --filter "Name=tag:Name,Values=aws-lc-ci-macos-arm-ec2-instance" "Name=instance-state-name,Values=running" --query Reservations[*].Instances[*].InstanceId --output text)"
```

Comment:
How does this search work? Isn't the host tagged `aws-lc-ci-macos-arm-dedicated-host` up in the mac CDK script?